glib/src/gstring.rs

```diff
@@ -102,6 +102,67 @@ impl GStr {
     pub fn as_c_str(&self) -> &CStr {
         unsafe { CStr::from_bytes_with_nul_unchecked(self.to_bytes_with_nul()) }
     }
+
+    #[doc(alias = "g_utf8_collate")]
+    #[inline]
+    pub fn collate(&self, other: &Self) -> Ordering {
+        unsafe { ffi::g_utf8_collate(self.as_ptr(), other.as_ptr()) }.cmp(&0)
+    }
+
+    #[doc(alias = "g_utf8_casefold")]
+    #[inline]
+    pub fn casefold(&self) -> GString {
+        let str = unsafe { ffi::g_utf8_casefold(self.as_ptr(), self.len() as isize) };
+
+        unsafe { from_glib_full(str) }
```

Comment:
```suggestion
        unsafe {
            let str = ffi::g_utf8_casefold(self.as_ptr(), self.len() as isize);
            from_glib_full(str)
        }
```

The whole block is unsafe, not just each of those smaller blocks on their own

Reply:
(same elsewhere)